requester/requester.go

```diff
@@ -161,14 +189,14 @@ func (r *Requester) ExecuteGraphql(query string, variables map[string]interface{
 		errMap := err.(map[string]interface{})
 		errorMessage := errMap["message"].(string)
 		if errMap["extensions"] == nil {
-			return nil, errors.New(errorMessage)
+			return nil, GraphQLError{Message: errorMessage}
 		}
 		extensions := errMap["extensions"].(map[string]interface{})
 		if extensions["error_name"] == nil {
-			return nil, errors.New(errorMessage)
+			return nil, GraphQLError{Message: errorMessage}
 		}
 		errorName := extensions["error_name"].(string)
-		return nil, errors.New(errorName + " - " + errorMessage)
+		return nil, GraphQLError{Message: errorName + ": " + errorMessage}
```

Comment:
It should be easy to distinguish errors with an `error_name` (which are almost all user exceptions) versus those without (which are probably our fault). Maybe a separate field in `GraphQLError` for the `error_name` (which I'd probably rather call "type")?